### DISCUSSION TITLE
error while downloading configs

### DISCUSSION THREAD

**Discussion by kverreth** [Dynatrace/dynatrace-configuration-as-code#447] [Q&A]
Hi

Trying to use the CLI 2.0 download function and running into the following error, it seems i am overlooking something. Is it an API token or an PAAS token you require?

`[xxxxxx(root):/home/kverreth/testDT] NEW_CLI=1 monaco download --downloadSpecificAPI alerting-profile -e /home/kverreth/testDT/environments.yaml
You are using the new CLI structure which is currently in Beta.`

`Please provide feedback here:
  https://github.com/dynatrace-oss/dynatrace-monitoring-as-code/issues/45.`

`We plan to make this CLI GA in version 2.0.0`

`2021-10-05 08:45:47 INFO  Dynatrace Monitoring as Code v1.6.0
2021-10-05 08:45:47 INFO  Creating base project name lat
2021-10-05 08:45:47 ERROR error retrieving token for enviroment lat environment variable TEST not found
2021-10-05 08:45:47 ERROR error while downloading configs for environment lat %!v(MISSING)`

The environments.yaml contains:

`lat:`
 `- name: "lat"`
`  - env-url: "https://https://zio347.dynatrace-managed.com/e/number"`
` - env-token-name: "TEST"`

API Token with full access is created on the environment.
![afbeelding](https://user-images.githubusercontent.com/91936346/135978364-52588858-84c3-4fe7-9331-c6e0dc116a99.png)

**Comment by Luke-Skelton** [comment]
Any update on resolving this issue, I'm experiencing something similar? Or were you able to resolve yourself?

**Reply by drqc** [reply]
Please make sure to run `export TEST=dt...yourtokenid` before running monaco download to make the dynatrace variable available to the console

**Reply by mdicss** [reply]
env-token-name gives the name of an environment variable containing the token, not the token itself.
It should be possible to add the token directly to the command line like this:
NEW_CLI=1 TEST=_token_ monaco download -environments=...
